### PR TITLE
Add fjord foundry V2 fees and volume

### PR DESCRIPTION
**Pull request by ggrievous3**

### CODE REVIEW

**Comment by llamatester** [comment]
The fjord-foundry adapter exports: 
        
 
 _______________________________________
    Dexs for 17/4/2024
    _______________________________________
    
    Version -> V2
    ---------
    ETHEREUM 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total volume: 269.57 M
    Daily volume: 2.44 M
    
    
    POLYGON 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total volume: 335.88 k
    
    
    ARBITRUM 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total volume: 1.01 M
    
    
    AVAX 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total volume: 236.33 k
    
    
    BASE 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total volume: 18.17 k
    
    
    BLAST 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total volume: 9.58 M
    
    
    BSC 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total volume: 0
    
    
    Done in 65.20s.

**Comment by llamatester** [comment]
The fjord-foundry adapter exports: 
        
 
 _______________________________________
    Fees for 17/4/2024
    _______________________________________
    
    Version -> V2
    ---------
    ETHEREUM 👇
    Backfill start time: 18/12/2023
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total fees: 10.67 M
    └─ Methodology: Fees collected from user trading fees
    Daily fees: 57.54 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 10.67 M
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    Daily revenue: 57.54 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    POLYGON 👇
    Backfill start time: 18/12/2023
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total fees: 18.69 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 18.69 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    ARBITRUM 👇
    Backfill start time: 18/12/2023
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total fees: 43.26 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 43.26 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    AVAX 👇
    Backfill start time: 18/12/2023
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total fees: 16.41 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 16.41 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    BASE 👇
    Backfill start time: 18/12/2023
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total fees: 2.26 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 2.26 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    BLAST 👇
    Backfill start time: 18/12/2023
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total fees: 407.32 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 407.32 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    BSC 👇
    Backfill start time: 18/12/2023
    Timestamp: 1713225600 (2024-04-16T00:00:00.000Z)
    Total fees: 98
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 98
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    Done in 65.12s.

**Comment by dtmkeng** [comment]
is there tvl adapter ?

**Comment by ggrievous3** [comment]
Not yet.

**Comment by ggrievous3** [comment]
We are in the process of launching our token and staking contracts. Then will add TVL adapter as well.

**Comment by dtmkeng** [comment]
volume is swap volume like uniswap ?

**Comment by ggrievous3** [comment]
> volume is swap volume like uniswap ?

yes @dtmkeng

**Comment by dtmkeng** [comment]
okay wait tvl adapter

**Comment by ggrievous3** [comment]
> okay wait tvl adapter

What do you mean?
We can't add fees generated and volume now?
The TVL could take a lot of time for us to get on mainnet. Because we are a launchpad and help all the protocols in raising money. 
You can see the website here [Fjord Foundry](https://www.fjordfoundry.com)

**Comment by dtmkeng** [comment]
yes wait tvl list

**Comment by dtmkeng** [comment]
you have tvl should list tvl first, is it hard ?

**Comment by ggrievous3** [comment]
we don't have TVL. That's what I'm saying. We don't rely on TVL on our protocol. Rather we help other protocols raise money.

**Comment by dtmkeng** [comment]
volume is agg not dexs like uni ?

**Comment by ggrievous3** [comment]
> volume is agg not dexs like uni ?

This is not agg vol. This volume is on our protocol.
https://app.v2.fjordfoundry.com/pools
You can see here all the volume is our own like uniswap.

**Comment by dtmkeng** [comment]
> > volume is agg not dexs like uni ?
> 
> This is not agg vol. This volume is on our protocol. https://app.v2.fjordfoundry.com/pools You can see here all the volume is our own like uniswap.

can't list volume list this

**Comment by ggrievous3** [comment]
> > > volume is agg not dexs like uni ?
> > 
> > 
> > This is not agg vol. This volume is on our protocol. https://app.v2.fjordfoundry.com/pools You can see here all the volume is our own like uniswap.
> 
> can't list volume list this

what do you mean can't list. why not ? This is our volume and not from anywhere else. We are a dex.

**Comment by dtmkeng** [comment]
plz remove volume and will list only fees 

> > > > volume is agg not dexs like uni ?
> > > 
> > > 
> > > This is not agg vol. This volume is on our protocol. https://app.v2.fjordfoundry.com/pools You can see here all the volume is our own like uniswap.
> > 
> > 
> > can't list volume list this
> 
> what do you mean can't list. why not ? This is our volume and not from anywhere else. We are a dex.

which tvl like uniswap ?

**Comment by ggrievous3** [comment]
> plz remove volume and will list only fees
> 
> > > > > volume is agg not dexs like uni ?
> > > > 
> > > > 
> > > > This is not agg vol. This volume is on our protocol. https://app.v2.fjordfoundry.com/pools You can see here all the volume is our own like uniswap.
> > > 
> > > 
> > > can't list volume list this
> > 
> > 
> > what do you mean can't list. why not ? This is our volume and not from anywhere else. We are a dex.
> 
> which tvl like uniswap ?

That's what I'm saying, our protocol doesn't depend on TVL. As soon as any protocol raises funds, they distribute their tokens. Hence we have volume and fees, but not TVL.

**Comment by dtmkeng** [comment]
> > plz remove volume and will list only fees
> > > > > > volume is agg not dexs like uni ?
> > > > > 
> > > > > 
> > > > > This is not agg vol. This volume is on our protocol. https://app.v2.fjordfoundry.com/pools You can see here all the volume is our own like uniswap.
> > > > 
> > > > 
> > > > can't list volume list this
> > > 
> > > 
> > > what do you mean can't list. why not ? This is our volume and not from anywhere else. We are a dex.
> > 
> > 
> > which tvl like uniswap ?
> 
> That's what I'm saying, our protocol doesn't depend on TVL. As soon as any protocol raises funds, they distribute their tokens. Hence we have volume and fees, but not TVL.

yes can't count volume like this

**Comment by ggrievous3** [comment]
> > > plz remove volume and will list only fees
> > > > > > > volume is agg not dexs like uni ?
> > > > > > 
> > > > > > 
> > > > > > This is not agg vol. This volume is on our protocol. https://app.v2.fjordfoundry.com/pools You can see here all the volume is our own like uniswap.
> > > > > 
> > > > > 
> > > > > can't list volume list this
> > > > 
> > > > 
> > > > what do you mean can't list. why not ? This is our volume and not from anywhere else. We are a dex.
> > > 
> > > 
> > > which tvl like uniswap ?
> > 
> > 
> > That's what I'm saying, our protocol doesn't depend on TVL. As soon as any protocol raises funds, they distribute their tokens. Hence we have volume and fees, but not TVL.
> 
> yes can't count volume like this

Like what ?

**Comment by dtmkeng** [comment]
no tvl

**Comment by dtmkeng** [comment]
if need dexs volume plz add tvl adapter

**Comment by 0xngmi** [comment]
when is data in your api updated?

**Comment by ggrievous3** [comment]
> when is data in your api updated?

It's updated in real-time @0xngmi . Whenever an API request is made we get the data since calculated from last time from the indexer and add it to the already calculated data.

File: dexs/fjord-foundry/index.ts [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    const totalVolume = chainData.stats
+        .filter(item => item.timestamp <= dayTimestamp)
+        .reduce((acc, { volume }) => acc + volume, 0)
+
+    const dailyVolume = chainData.stats
+        .find(dayItem => dayItem.timestamp === dayTimestamp)?.volume
+
+    return {
+        timestamp: dayTimestamp,
```
Comment (0xngmi) [inline dexs/fjord-foundry/index.ts]: pls remove these timestamps and getTimestampAtStartOfPreviousDayUTC then

Comment (ggrievous3) [reply dexs/fjord-foundry/index.ts]: I saw this on the example in the docs. Without this will you be able to get the historical data as well day-wise ?

Comment (ggrievous3) [reply dexs/fjord-foundry/index.ts]: ok, i've removed `timestamp`. Can you check.

**Comment by llamatester** [comment]
The fjord-foundry adapter exports: 
        
 
 _______________________________________
    Dexs for 18/4/2024
    _______________________________________
    
    Version -> V2
    ---------
    ETHEREUM 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Total volume: 277.10 M
    Daily volume: 7.53 M
    
    
    POLYGON 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Total volume: 335.88 k
    
    
    ARBITRUM 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Total volume: 1.01 M
    
    
    AVAX 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Total volume: 236.33 k
    
    
    BASE 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Total volume: 18.17 k
    
    
    BLAST 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Total volume: 9.58 M
    
    
    BSC 👇
    Backfill start time: 18/12/2023
    NO METHODOLOGY SPECIFIED
    Total volume: 0
    
    
    Done in 81.72s.

**Comment by llamatester** [comment]
The fjord-foundry adapter exports: 
        
 
 _______________________________________
    Fees for 18/4/2024
    _______________________________________
    
    Version -> V2
    ---------
    ETHEREUM 👇
    Backfill start time: 18/12/2023
    Total fees: 10.82 M
    └─ Methodology: Fees collected from user trading fees
    Daily fees: 150.58 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 10.82 M
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    Daily revenue: 150.58 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    POLYGON 👇
    Backfill start time: 18/12/2023
    Total fees: 18.69 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 18.69 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    ARBITRUM 👇
    Backfill start time: 18/12/2023
    Total fees: 43.26 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 43.26 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    AVAX 👇
    Backfill start time: 18/12/2023
    Total fees: 16.41 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 16.41 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    BASE 👇
    Backfill start time: 18/12/2023
    Total fees: 2.26 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 2.26 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    BLAST 👇
    Backfill start time: 18/12/2023
    Total fees: 407.32 k
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 407.32 k
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    BSC 👇
    Backfill start time: 18/12/2023
    Total fees: 98
    └─ Methodology: Fees collected from user trading fees
    Total revenue: 98
    └─ Methodology: Revenue is 100% fee of each swap which goes to treasury
    
    
    Done in 63.29s.

File: dexs/fjord-foundry/index.ts (hunk 1 above)
Comment (ggrievous3) [reply dexs/fjord-foundry/index.ts]: any update on this @0xngmi ?

**Pull request merged**